### PR TITLE
Validation of transactions ordinal chain during block acceptance

### PR DESCRIPTION
**Pull request by kpudlik**

### CODE REVIEW

**Review by marcinwadon — DISMISSED** [review]

File: src/main/scala/org/constellation/checkpoint/CheckpointBlockValidator.scala [hunk 1]
```diff
@@ -160,6 +160,21 @@ class CheckpointBlockValidator[F[_]: Sync](
   ): F[ValidationResult[List[Transaction]]] =
     t.toList.traverse(validateTransaction).map(x => x.map(_.map(List(_)))).map(_.combineAll)
 
+  def validateTransactionOrdinalChain(t: Iterable[Transaction]): ValidationResult[List[Transaction]] = {
```
Comment (tbekas) [inline src/main/scala/org/constellation/checkpoint/CheckpointBlockValidator.scala:163]: There's already a logic that does this validation. I would suggest extracting it to external class and use it in both places. `CheckpointAcceptanceService#isCheckpointBlockAllowedForAcceptance`

Comment (kpudlik) [reply src/main/scala/org/constellation/checkpoint/CheckpointBlockValidator.scala:163]: @tbekas Your validation checks if `hashes` form a chain via `lastTxRef` but here I check if `ordinals` are correct. I think we can use both validations anyway.

Comment (tbekas) [reply src/main/scala/org/constellation/checkpoint/CheckpointBlockValidator.scala:163]: We talked with @kpudlik offline, but just for the record, the existing validation in `CheckpointAcceptanceService#isCheckpointBlockAllowedForAcceptance` checks both `prevHash` and `ordinal` of any transaction that's in the chain.

Comment (kpudlik) [reply src/main/scala/org/constellation/checkpoint/CheckpointBlockValidator.scala]: Yep, I can confirm that @tbekas is right. I have extracted @tbekas's logic and reused in all the places where it was duplicated.

**Review by tbekas — APPROVED** [review]
Looks good. You will likely need to rebase your branch.

**Review by marcinwadon — APPROVED** [review]

**Review by marcinwadon — APPROVED** [review]

**Pull request merged**